hudi-hadoop-common/src/main/java/org/apache/parquet/avro/AvroSchemaConverterWithTimestampNTZ.java

```diff
@@ -267,121 +271,123 @@ private Type convertUnion(String fieldName, Schema schema, Type.Repetition repet
+              return HoodieSchema.create(HoodieSchemaType.FLOAT);
             }
             @Override
-            public Schema convertDOUBLE(PrimitiveTypeName primitiveTypeName) {
-              return Schema.create(Schema.Type.DOUBLE);
+            public HoodieSchema convertDOUBLE(PrimitiveTypeName primitiveTypeName) {
+              return HoodieSchema.create(HoodieSchemaType.DOUBLE);
             }
             @Override
-            public Schema convertFIXED_LEN_BYTE_ARRAY(PrimitiveTypeName primitiveTypeName) {
+            public HoodieSchema convertFIXED_LEN_BYTE_ARRAY(PrimitiveTypeName primitiveTypeName) {
               if (annotation instanceof LogicalTypeAnnotation.UUIDLogicalTypeAnnotation) {
-                return Schema.create(Schema.Type.STRING);
+                return HoodieSchema.createUUID();
               } else {
                 int size = parquetType.asPrimitiveType().getTypeLength();
-                return Schema.createFixed(parquetType.getName(), null, null, size);
+                return HoodieSchema.createFixed(parquetType.getName(), null, null, size);
               }
             }
             @Override
-            public Schema convertBINARY(PrimitiveTypeName primitiveTypeName) {
+            public HoodieSchema convertBINARY(PrimitiveTypeName primitiveTypeName) {
               if (annotation instanceof LogicalTypeAnnotation.StringLogicalTypeAnnotation ||
                   annotation instanceof  LogicalTypeAnnotation.EnumLogicalTypeAnnotation) {
-                return Schema.create(Schema.Type.STRING);
+                return HoodieSchema.create(HoodieSchemaType.STRING);
               } else {
-                return Schema.create(Schema.Type.BYTES);
+                return HoodieSchema.create(HoodieSchemaType.BYTES);
               }
             }
           });
 
-      LogicalType logicalType = convertLogicalType(annotation);
-      if (logicalType != null && (!(annotation instanceof LogicalTypeAnnotation.DecimalLogicalTypeAnnotation) ||
-          parquetPrimitiveTypeName == BINARY ||
-          parquetPrimitiveTypeName == FIXED_LEN_BYTE_ARRAY)) {
-        schema = logicalType.addToSchema(schema);
-      }
-
       return schema;
 
     } else {
       GroupType parquetGroupType = parquetType.asGroupType();
       LogicalTypeAnnotation logicalTypeAnnotation = parquetGroupType.getLogicalTypeAnnotation();
       if (logicalTypeAnnotation != null) {
-        return logicalTypeAnnotation.accept(new LogicalTypeAnnotation.LogicalTypeAnnotationVisitor<Schema>() {
+        return logicalTypeAnnotation.accept(new LogicalTypeAnnotation.LogicalTypeAnnotationVisitor<HoodieSchema>() {
           @Override
-          public java.util.Optional<Schema> visit(LogicalTypeAnnotation.ListLogicalTypeAnnotation listLogicalType) {
+          public java.util.Optional<HoodieSchema> visit(LogicalTypeAnnotation.ListLogicalTypeAnnotation listLogicalType) {
```

Comment:
While we're here, let's import `Optional` to shorten the line lengths in this file